include/swift/Runtime/Config.h

```diff
@@ -38,7 +38,7 @@
 #ifndef SWIFT_USE_SWIFTCALL
 // Clang doesn't support mangling functions with the swiftcall attribute
 // on Windows and crashes during compilation: http://bugs.llvm.org/show_bug.cgi?id=32000
-#if (__has_attribute(swiftcall) || defined(__linux__)) && !defined(_WIN32)
+#if __has_attribute(swiftcall) && !defined(_WIN32)
```

Comment:
It's now #including ABI/HeapObject.h, but I'm not sure why that's a legit thing to be including, either.  Honestly, I'm not sure why that header is in the ABI directory instead of the Runtime directory: it's declaring the layout of target structures.

I think what Joe and I are saying is that this should not be determined in the compiler based on whether the target machine supports the CC, but rather some more semantic database of target configurations which Runtime/Config.h could also cue off of.  That database should be exhaustive over our targets (arch + OS pairs).

If we need to TableGen that database to make it maintainable, so be it.